slick/src/main/scala/slick/dbio/DBIOAction.scala

```diff
@@ -554,7 +561,7 @@ object SynchronousDatabaseAction {
     * fused action can fail with a `ClassCastException` during evaluation. */
   private[slick] def fuseUnsafe[R, S <: NoStream, E <: Effect](a: DBIOAction[R, S, E]): DBIOAction[R, S, E] = {
     a match {
-      case FlatMapAction(base: SynchronousDatabaseAction[_, _, _, _], f, ec) if ec eq DBIO.sameThreadExecutionContext =>
+      case FlatMapAction(base: SynchronousDatabaseAction[_, _, _, _], Vector(f), ec) if ec eq DBIO.sameThreadExecutionContext =>
```

Comment:
I had a closer look, in general it doesn't seem safe to fuse flatMap as we are not able to be sure if the result is fusable, and there are non-database computations going on in the function as well. The [documentation](http://slick.lightbend.com/doc/3.1.0/dbio.html#transactions-and-pinned-sessions) seems to support this as well.

It looks like the only time fuseUnsafe is used internally with transactions? In the cases where this code is triggered during tests it's always with a single chained action.